### PR TITLE
Python 3.8.4 release breaks our builds

### PR DESCRIPTION
**Pull request by potiuk**
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

**Review by feluelle — COMMENTED** [review]
Can you also add it to the README https://github.com/apache/airflow#additional-notes-on-python-version-requirements ? :)

**Comment by potiuk** [comment]
> Can you also add it to the README https://github.com/apache/airflow#additional-notes-on-python-version-requirements ? :)

I will - but I think we need to know the root cause before. It might be something that will be fixed in sqlalchemy - in which case it might work fine with 3.8.4 when they fix it.

File: scripts/ci/libraries/_initialization.sh [hunk 1]
```diff
@@ -21,6 +21,13 @@ function initialize_common_environment {
     # default python Major/Minor version
     PYTHON_MAJOR_MINOR_VERSION=${PYTHON_MAJOR_MINOR_VERSION:="3.6"}
 
+    PYTHON_BASE_IMAGE_VERSION=${PYTHON_MAJOR_MINOR_VERSION}
+    if [[ ${PYTHON_BASE_IMAGE_VERSION} == "3.8" ]]; then
+        # Temporary fix for Python 3.8.4 failing our tests
+        # TODO: Remove after it gets fixed
+        PYTHON_BASE_IMAGE_VERSION="3.8.3"
```
Comment (feluelle) [inline scripts/ci/libraries/_initialization.sh:28]: Can we `echo` this?

Comment (potiuk) [reply scripts/ci/libraries/_initialization.sh:28]: Good idea. Added.

**Review by feluelle — APPROVED** [review]

**Pull request merged**